src/main/java/org/skriptlang/skript/bukkit/recipes/elements/CondDiscoveredRecipes.java

```diff
@@ -50,13 +50,13 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 	@Override
 	public boolean check(Event event) {
 		Recipe[] recipes = exprRecipe.getArray(event);
-		return exprPlayer.check(event, player -> {
-			return SimpleExpression.check(recipes, recipe -> {
+		return exprPlayer.check(event, player ->
+			SimpleExpression.check(recipes, recipe -> {
 				if (!(recipe instanceof Keyed recipeKey))
-					return false;
+					return isNegated();
 				return player.hasDiscoveredRecipe(recipeKey.getKey());
-			}, isNegated(), exprRecipe.getAnd());
-		});
+			}, isNegated(), exprRecipe.getAnd())
+		);
```

Comment:
are you sure this is correct? Using isNegated twice generally means it won't actually work, since it should only be applied once.

Reply:
How do you think I should go about handling recipes that on the off chance are not Keyed?
I just assumed to always have it false

Reply:
@sovdeeth

Reply:
based on precedent in https://github.com/SkriptLang/Skript/pull/7038, the previous implementation was fine